specification/cognitiveservices/OpenAI.Inference/models/completions/azure_content_filtering.tsp

```diff
@@ -0,0 +1,175 @@
+
+  @doc("""
+  Describes language related to physical actions intended to hurt, injure, damage, or 
+  kill someone or something; describes weapons, etc.
+  """)
+  violence?: ContentFilterResult;
+
+  @doc("""
+  Describes language attacks or uses that include pejorative or discriminatory language 
+  with reference to a person or identity group on the basis of certain differentiating 
+  attributes of these groups including but not limited to race, ethnicity, nationality,
+  gender identity and expression, sexual orientation, religion, immigration status, ability
+  status, personal appearance, and body size.
+  """)
+  hate?: ContentFilterResult;
+
+  @doc("""
+  Describes language related to physical actions intended to purposely hurt, injure,
+  or damage one’s body, or kill oneself.
+  """)
+  @projectedName("json", "self_harm")
+  selfHarm?: ContentFilterResult;
+
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @doc("Describes whether profanity was detected.")
+  profanity?: ContentFilterDetectionResult;
+
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @projectedName("json", "custom_blocklists")
+  @doc("Describes detection results against configured custom blocklists.")
+  customBlocklists?: ContentFilterBlocklistIdResult[];
+
+  @doc("""
+  Describes an error returned if the content filtering system is
+  down or otherwise unable to complete the operation in time.
+  """)
+  error?: Azure.Core.Foundations.Error;
+};
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("Information about content filtering evaluated against input data to Azure OpenAI.")
+model ContentFilterResultDetailsForPrompt {
+  ...ContentFilterResultsBase;
+
+  @doc("Whether a jailbreak attempt was detected in the prompt.")
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  jailbreak?: ContentFilterDetectionResult;
+}
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("Information about content filtering evaluated against generated model output.")
+model ContentFilterResultsForChoice {
+  ...ContentFilterResultsBase;
+
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @projectedName("json", "protected_material_text")
+  @doc("Information about detection of protected text material.")
+  protectedMaterialText?: ContentFilterDetectionResult;
+
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @projectedName("json", "protected_material_code")
+  @doc("Information about detection of protected code material.")
+  protectedMaterialCode?: ContentFilterCitedDetectionResult;
+}
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("""
+Content filtering results for a single prompt in the request.
+""")
+model ContentFilterResultsForPrompt {
+  @doc("The index of this prompt in the set of prompt results")
+  @projectedName("json", "prompt_index")
+  promptIndex: int32;
+
+  @doc("Content filtering results for this prompt")
+  @projectedName("json", "content_filter_results")
+  contentFilterResults?: ContentFilterResultDetailsForPrompt;
```

Comment:
Should this be required? I wonder why would we get an entry with a prompt index but no content filter results.Should this be required? I wonder why would we get an entry with a prompt index but no content filter results.

Reply:
Good question. Reviewing the .json document, [it doesnlt look like EITHER of the two properties are marked as required](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-12-01-preview/inference.json#L1152), which is particularly strange to imagine. I'll follow up to see if there's a reason for this but I'm inclined to agree with your assessment and make these both required until proven otherwise; it seems like it's just a spec omission that we can correct.